python/iceberg/api/expressions/literals.py

```diff
@@ -243,6 +260,9 @@ def to(self, type_var):
                                       .quantize(Decimal("." + "".join(["0" for i in range(1, type_var.scale)]) + "1"),
                                                 rounding=ROUND_HALF_UP))
 
+    def type_id(self):
```

Comment:
I wonder if it would be cleaner to add the `type_id` as a `__init__` param in `BaseLiteral`?

Reply:
That makes the most sense to me.  Since it will cause a number of changes in unrelated files, what do you think about opening an issue on it, and handling in a separate PR?

Reply:
Why would it affect other files? To me it would just make this diff smaller: add `type_id` parameter to `__init__` and return it instead of the `NotImplementedError` in `BaseLiteral.type_id`, then we don't have to implement `def type_id(self):` in every subclass.

But yeah, not overly precious about it. Can def be done in a followup.

Reply:
I think I misunderstood what you were proposing.  Let me walk that back, haha.  You're just saying in the child-class pass into the super constructor the type_id of the child and then just implement a single type_id attr in the BaseLiteral.  I agree that does seem better to me and is appropriate to handle within this PR.  

I'm also going to go ahead and finish my coffee before reading any more PRs.

Reply:
> I'm also going to go ahead and finish my coffee before reading any more PRs.

:rofl:

Reply:
👍  thanks for the comments. SGTM